src/main/kotlin/com/openlattice/data/DataGraphService.kt

```diff
@@ -375,8 +380,13 @@ open class DataGraphService(
                 .asMap(associations)
                 .forEach {
                     val entitySetId = it.key
-                    val entities = it.value.map { it.data }
-                    val (ids, entityWrite) = createEntities(entitySetId, entities, authorizedPropertiesByEntitySetId[entitySetId]!!)
+
+                    // check entity types of associations before creation
+                    checkAssociationEntityTypes(entitySetId, it.value)
```

Comment:
This is better but still loads the whole AssociationTypes one at a time. You could add a ```getAssociationTypes( Set<UUID> associationTypeIds )``` to EdmManager and then call it ```getAssociationTypes(associations.keySet())```. Then make checkAssociationEntityTypes take the association type instead of the entitySetId.